src/platforms/pumpfun/event_parser.py

```
@@ -0,0 +1,485 @@
+        self._create_instruction_discriminator = struct.unpack("<Q", self._create_instruction_discriminator_bytes)[0]
+        
+        logger.info("Pump.Fun event parser initialized with IDL-based event and instruction parsing")
+        logger.info(f"CreateEvent discriminator: {self._create_event_discriminator_bytes.hex()}")
+        logger.info(f"create instruction discriminator: {self._create_instruction_discriminator_bytes.hex()}")
+    
+    @property
+    def platform(self) -> Platform:
+        """Get the platform this parser serves."""
+        return Platform.PUMP_FUN
+    
+    def parse_token_creation_from_logs(
```

Ruff (0.12.2)

`parse_token_creation_from_logs` is too complex (17 > 10)

(C901)